src/storage/src/monitor/local_metrics.rs

```diff
@@ -89,9 +117,19 @@ impl StoreLocalStatistic {
                 .bloom_filter_might_positive_counts
                 .inc_by(self.bloom_filter_might_positive_count);
         }
-        let t = self.remote_io_time.load(Ordering::Relaxed) as f64;
-        if t > 0.0 {
-            metrics.remote_read_time.observe(t / 1000.0);
+        let remote_io_time = self.remote_io_time.load(Ordering::Relaxed) as f64;
+        if remote_io_time > 0.0 {
+            metrics.remote_read_time.observe(remote_io_time / 1000.0);
+        }
+
+        self.reported = true;
+    }
+}
+
+impl Drop for StoreLocalStatistic {
```

Comment:
Is `Drop`  panic only for test  and use reported to avoid duplicated report ?

Reply:
The last commit is only to stash my workspace. 🤣 I already marked this PR as draft to avoid bother other reviewers. The related problem is discussed and fixed in #4723 .